modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/OpenAPIResourceImpl.java

```diff
@@ -174,27 +173,59 @@ else if (Objects.equals(parameterName, "relatedObjectEntryId")) {
 				});
 		}
 
-		return new PathItem() {
+		return new Operation() {
 			{
-				put(
-					new Operation() {
-						{
-							operationId(
-								StringBundler.concat(
-									"put",
-									_currentObjectDefinition.getShortName(),
-									StringUtil.upperCaseFirstLetter(
-										objectRelationship.getName()),
-									relatedObjectDefinition.getShortName()));
-							parameters(new ArrayList<>(parameters.values()));
-							responses(operation.getResponses());
-							tags(operation.getTags());
-						}
-					});
+				operationId(
+					StringBundler.concat(
+						httpMethod, _currentObjectDefinition.getShortName(),
+						StringUtil.upperCaseFirstLetter(
+							objectRelationship.getName()),
+						relatedObjectDefinition.getShortName()));
+				parameters(new ArrayList<>(parameters.values()));
+				responses(operation.getResponses());
+				tags(operation.getTags());
 			}
 		};
 	}
 
+	private PathItem _createPathItem(
+		ObjectRelationship objectRelationship, PathItem pathItem,
+		ObjectDefinition relatedObjectDefinition) {
+
+		Map<PathItem.HttpMethod, Operation>
+			relationshipHttpMethodOperationsMap = pathItem.readOperationsMap();
+
+		Operation relationshipOperation =
+			relationshipHttpMethodOperationsMap.get(PathItem.HttpMethod.GET);
+
+		if (relationshipOperation != null) {
+			PathItem customPathItem = new PathItem();
+
+			customPathItem.get(
+				_createCustomRelationshipOperation(
+					"get", pathItem.getGet(), relatedObjectDefinition,
+					objectRelationship));
+
+			return customPathItem;
+		}
+
+		relationshipOperation = relationshipHttpMethodOperationsMap.get(
+			PathItem.HttpMethod.PUT);
+
+		if (relationshipOperation != null) {
+			PathItem customPathItem = new PathItem();
+
+			customPathItem.put(
+				_createCustomRelationshipOperation(
+					"put", pathItem.getPut(), relatedObjectDefinition,
+					objectRelationship));
+
+			return customPathItem;
+		}
+
+		return new PathItem();
```

Comment:
Is possible this case?

Reply:
It should not be possible as in the `rest-openapi` there are only GET and PUT endpoints for relationships. But, as I'm not 100% sure that there will not be a new endpoint or there be a corner case that does not use GET or PUT, I prefer to apply the defensive programing to avoid the crush of the Object API 😁